project/src/main/kotlin/org/ionproject/core/programme/representations/ProgrammeListRepr.kt

```diff
@@ -0,0 +1,34 @@
+package org.ionproject.core.programme.representations
+
+import org.ionproject.core.common.*
+import org.ionproject.core.common.modelInterfaces.IProgramme
+import org.springframework.http.HttpMethod
+import java.net.URI
+
+
+fun programmesListRepr(programmes: List<IProgramme>) = SirenBuilder()
+        .klass("collection", "programme")
+        .entities(programmes.map {
+            buildSubentities(it)
+        })
+        .action(
+                Action(
+                        name = "add-programme", title = "Add Programme", method = HttpMethod.POST,
+                        href = URI(PROGRAMMES_PATH), type = JSON_MEDIA_TYPE,
+                        fields = listOf(
+                                Field(name = "ProgrammeAcr", type = "text"),
+                                Field(name = "TermSize", type = "number")
+                        )
+                )
+        ).link("self", href = URI(PROGRAMMES_PATH))
+        .toSiren()
```

Comment:
I feel like these OutputModel functions should be inside a class and not be global. Maybe have a ProgrammeTransformer class that is injected into the ProgrammeController

Reply:
@JorgePBrown there is no advantage in doing that, only making the code harder to read.

Reply:
How is there no advantage and how does it make it harder to read? Why not make the URIs just global constants aswell then instead of being in the URI object?

Reply:
Because each representation is different from each other, I don't think making a class and grouping all of them makes sense. The URI object is useful because of the UriTemplate and you don't have to manually go in the representation and write something like ".../courses/${course.id}/..." but just use the object like this: Uri.forCourse(course.id) .
If you did the siren representations wouldn't be consistent and in some cases you would use the constant and in others write the full link.